### PR TITLE
Feature 137 save portfolio scr

### LINKED ISSUE #137: Portfolio screen should not reload every time it is accessed

### PR DESCRIPTION
**Pull request by Valmetti**
resolves #137

### CODE REVIEW

File: src/components/providers/portfolio-provider.tsx [hunk 1]
```diff
@@ -59,24 +67,63 @@ const PortfolioProvider: React.FC = ({ children }) => {
...
+  useEffect(() => {
+    loadHistoryValues();
+  }, [ statisticsLoaderParams ]);
+
+  /**
+   * Sets the history loader interval
+   */
+  useInterval(() => {
+    loadHistoryValues();
+  }, 10000);
```
Comment (platonovsimeon) [inline src/components/providers/portfolio-provider.tsx:107]:
```suggestion
  }, 3600000);
```

**Review by platonovsimeon — CHANGES_REQUESTED** [review]

File: src/components/providers/portfolio-provider.tsx [hunk 2]
```diff
@@ -59,24 +67,63 @@ const PortfolioProvider: React.FC = ({ children }) => {
...
+   * Loads history values
+   */
+  const loadHistoryValues = async () => {
+    try {
+      if (!statisticsLoaderParams) {
+        return;
+      }
+
+      const { effectivePortfolios, range } = statisticsLoaderParams;
+      const { startDate, endDate } = DateUtils.getDateFilters(range);
```
Comment (jarinykanen) [inline src/components/providers/portfolio-provider.tsx:80]: Move things that do not throw errors outside try block

File: src/components/screens/portfolio/statistics-screen.tsx [hunk 3]
```diff
@@ -210,6 +230,16 @@ const StatisticsScreen: React.FC = () => {
     );
   };
 
+  /**
+   * Handles updating selected range
+   * 
+   * @param range
```
Comment (jarinykanen) [inline src/components/screens/portfolio/statistics-screen.tsx:236]:
```suggestion
   * @param range new range
```

File: src/components/screens/portfolio/statistics-screen.tsx [hunk 4]
```diff
@@ -229,7 +259,7 @@ const StatisticsScreen: React.FC = () => {
         <ChartRangeSelector
           selectedRange={ selectedRange }
           loading={ loading }
-          onDateRangeChange={ setSelectedRange }
+          onDateRangeChange={ range => onRangeUpdate(range) }
```
Comment (jarinykanen) [inline src/components/screens/portfolio/statistics-screen.tsx:262]:
```suggestion
          onDateRangeChange={ onRangeUpdate }
```

**Review by jarinykanen — APPROVED** [review]

File: src/components/providers/portfolio-provider.tsx (hunk 2 above)
Comment (platonovsimeon) [reply src/components/providers/portfolio-provider.tsx]: Fixed

**Pull request merged**